### PR TITLE
feat(rest): added create-offer endpoint and tests

### PR DESCRIPTION
**Pull request by rapaktech**
Signed-off-by: Jim Ezesinachi <jim@animo.id>

### CODE REVIEW

File: packages/rest/src/controllers/credentials/CredentialController.ts [hunk 1]
```diff
@@ -157,6 +158,31 @@ export class CredentialController extends Controller {
...
+   */
+  @Example<CredentialExchangeRecordProps>(CredentialExchangeRecordExample)
+  @Post('/create-offer')
+  public async createOffer(
+    @Body() options: CreateOfferOptions,
+    @Res() internalServerError: TsoaResponse<500, { message: string }>
+  ) {
+    try {
+      const offer = await this.agent.credentials.createOffer(options)
+      await this.agent.oob.createInvitation({ messages: [offer?.message] })
```
Comment (TimoGlastra) [inline packages/rest/src/controllers/credentials/CredentialController.ts:176]: Ah wait, this shouldn't call create invitation. There's multiple ways to create the invitation, and also a legacy way. So what we've done is first call create offer endpoint, and you the. Manually need to call create (legacy-connection less) invitation endpoint

[the remark this answers is not in the document]
Comment (rapaktech) [reply packages/rest/src/controllers/credentials/CredentialController.ts:176]: I wasn't sure if it was

Comment (rapaktech) [reply packages/rest/src/controllers/credentials/CredentialController.ts]: Okay

Comment (rapaktech) [reply packages/rest/src/controllers/credentials/CredentialController.ts]: When you say endpoint, you mean the method, correct?

**Review by TimoGlastra — APPROVED** [review]

**Pull request merged**